### PR TITLE
SB-748: Make the cron tests extend the TestCaseWithArtifactGenerationWithIndexing class and create/remove repositories properly during initialization/cleanup 

### PR DESCRIPTION
**Pull request by kate-novik**

### CODE REVIEW

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/AbstractLayoutProvider.java [hunk 1]
```diff
@@ -442,7 +443,8 @@ public FileVisitResult postVisitDirectory(Path dir,
                                                           IOException exc)
                         throws IOException
                 {
-                    Files.delete(dir);
+//                    Files.delete(dir);
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/AbstractLayoutProvider.java]: @kate-novik : Please, remove this.

**Comment by carlspring** [comment]
@kate-novik : Could you please fix the issues reported from Sonar which make sense to you? Especially 1...3, as they are marked as critical and will cause the Sonar checks to fail.

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/AbstractLayoutProvider.java (hunk 1 above)
Comment (kate-novik) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/AbstractLayoutProvider.java]: ok, i removed.

**Comment by kate-novik** [comment]
I fixed the issues reported from Sonar.

**Pull request merged**